libbeat/api/server.go

```diff
@@ -33,13 +33,15 @@ import (
 // Server takes care of correctly starting the HTTP component of the API
 // and will answer all the routes defined in the received ServeMux.
 type Server struct {
-	log        *logp.Logger
-	mux        *http.ServeMux
-	l          net.Listener
-	config     Config
-	wg         sync.WaitGroup
-	mutex      sync.Mutex
-	httpServer *http.Server
+	log         *logp.Logger
+	mux         *http.ServeMux
+	l           net.Listener
+	config      Config
+	wg          sync.WaitGroup
+	mutex       sync.Mutex
+	httpServer  *http.Server
+	startCalled bool
+	stopCalled  bool
```

Comment:
(Completely optional suggestion) Given that these are clearly ordered, perhaps a single enum with states inactive, started, stopped will make it easier to read?